bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/asset/HierarchicalElement.java

```diff
@@ -120,4 +124,26 @@ default void visitAllFiles(CheckedConsumer<HierarchicalElement> visitor,
     default void visitAllFiles(CheckedConsumer<HierarchicalElement> visitor) throws Exception {
         visitAllFiles(visitor, HierarchicalElement::getFolderChildren, HierarchicalElement::getFileChildren);
     }
-}
\ No newline at end of file
+
+
+    class UriHelper {
+        static String SFTP_URL_ENCODING = "utf-8";
+
+        private UriHelper(){}
+
+        static String encodeUriParts(final String uri) throws UnsupportedEncodingException {
+            String[] uriParts = uri.split("/");
+
+            //path parts started from index 3
+            for (int i = 3; i < uriParts.length; i++) {
```

Comment:
Would be nice to replace a magic number 3 with a constant

Reply:
But then it would lose its magic!